### PR TITLE
Contracts/test/state transition testing

### PR DESCRIPTION
**Pull request by josojo**
added all state transition tests based on

https://drive.google.com/drive/folders/0ByHhiGx-ltJZczhjZHhHeGpHcHM

only further cleaning needs to be required

contains small adaptions for the smart contract, but only ones for better testing - no logic impact

### CODE REVIEW

**Review by W3stside — COMMENTED** [review]
@josojo can you please change the base branch to the DX-2.0-No-Frontend? or was there a specific reason why you chose `tulipTesting` again?

File: test/stateTransitions.js [hunk 1]
```diff
@@ -56,7 +53,7 @@ const checkState = async (auctionIndex, auctionStart, sellVolumesCurrent, sellVo
 
 const getState = async (contracts, ST, BT) => {
```
Comment (W3stside) [inline test/stateTransitions.js]: @josojo Why are you passing in `contracts` here? It is in the upper scope but most importantly you are not using it in the actual fn block. Go ahead and remove this unless there's something I'm not seeing ;)

EDIT: if you do remove `contracts` from the fn signature, be sure to change all the dependent fns! 😄 I just did a simple `ctrl + f` --> `getState(contracts, ` <-- mind the last space and replaced w/ `getState(`

File: test/stateTransitions.js [hunk 2]
```diff
@@ -56,7 +53,7 @@ const checkState = async (auctionIndex, auctionStart, sellVolumesCurrent, sellVo
...
   const auctionStart = (await dx.getAuctionStart.call(eth.address, gno.address)).toNumber()
-  if (auctionStart == 1) { return 5 }
+  if (auctionStart === 1) { return 5 }
   const auctionIndex = await getAuctionIndex()
   console.log('at least here')
   let numP
```
Comment (W3stside) [inline test/stateTransitions.js]: @josojo lines 59 - 64 can also be written as:
```
let nump, denp, numPP, denPP, numBasedOnVolume, denBasedOnVolume // just stylistic :D
```

**Comment by josojo** [comment]
changed the branch for merging.

**Pull request merged**